Library/Formula/mandoc.rb

```diff
@@ -0,0 +1,74 @@
+class Mandoc < Formula
+  homepage "http://mdocml.bsd.lv"
+  url "http://mdocml.bsd.lv/snapshots/mdocml-1.13.3.tar.gz"
+  sha256 "23ccab4800d50bf4c327979af5d4aa1a6a2dc490789cb67c4c3ac1bd40b8cad8"
+
+  head "anoncvs@mdocml.bsd.lv:/cvs", :module => "mdocml", :using => :cvs
+
+  depends_on "sqlite"
+
+  option "without-db", "Only install the mandoc/demandoc utilities."
+  option "without-cgi", "Don't build man.cgi (and extra CSS files)."
+
+  def install
+    localconfig = [
+
+      # Sane prefixes.
+      "PREFIX=#{prefix}",
+      "INCLUDEDIR=#{include}",
+      "LIBDIR=#{lib}",
+      "MANDIR=#{man}",
+      "WWWPREFIX=#{prefix}/var/www",
+      "EXAMPLEDIR=#{share}/examples",
+
+      # Executable names, where utilites would be replaced/duplicated.
+      # The mdocml versions of the utilities are definitely *not* ready
+      # for prime-time on Darwin, though some changes in HEAD are promising.
+      # The "bsd" prefix (like bsdtar, bsdmake) is more informative than "m".
+      "BINM_MAN=bsdman",
+      "BINM_APROPOS=bsdapropos",
+      "BINM_WHATIS=bsdwhatis",
+      "BINM_MAKEWHATIS=bsdmakewhatis",	# default is "makewhatis".
+
+      # These are names for *section 7* pages only. Several other pages are
+      # prefixed "mandoc_", similar to the "groff_" pages.
+      "MANM_MAN=man",
+      "MANM_MDOC=mdoc",
+      "MANM_ROFF=mandoc_roff", # This is the only one that conflicts (groff).
+      "MANM_EQN=eqn",
+      "MANM_TBL=tbl",
+
+      "OSNAME='Mac OS X #{MacOS.version}'", # Bottom corner signature line.
+
+      # Not quite sure what to do here. The default ("/usr/share", etc.) needs
+      # sudoer privileges, or will error. So just brew's manpages for now?
+      "MANPATH_DEFAULT=#{HOMEBREW_PREFIX}/share/man",
```

Comment:
The `[bsd]makewhatis` utility traverses the manpath and writes a `mandoc.db` in each `man` directory. This database (a modern replacement for `whatis` files, e.g. `/usr/share/man/whatis`) is then read by `[bsd]apropos`, `[bsd]whatis`, and `[bsd]man`. So to answer your question, its both reading and writing, and this is also why there's a privileges problem, since it wants to write to `/usr/share/man/mandoc.db`, etc. Presently, the mandoc `man` utility depends on this file's existence, but upstream its been updated to fall back to manually traversing the file hierarchy, so it will should be more usable in the next release.